File: solutions/TypeScript/[0080_Medium] Remove Duplicates from Sorted Array II/removeDuplicates.ts
function removeDuplicates(nums: number[]): number {
  let { length } = nums;
  let current = 2;

  while (current < length) {
    if (nums[current - 1] === nums[current] && nums[current - 2] === nums[current]) {
      for (let i = current; i < length - 1; i++) {
        nums[i] = nums[i + 1];
      }
      length--;
    } else {
      current++;
    }
  }

  return length;
}

File: solutions/Java/[0026_Easy] Remove Duplicates from Sorted Array/removeDuplicates.java
class Solution {
  public int removeDuplicates(int[] nums) {
    int fast = 1;
    int slow = 1;

    while (fast < nums.length) {
      if (nums[fast] != nums[fast - 1]) {
        nums[slow] = nums[fast];
        slow++;
      }

      fast++;
    }

    return slow;
  }
}
